docs/testplan/DPU-test-plan.md

```diff
@@ -0,0 +1,863 @@
+
+FPDs
+    RP0/info.0                          0.8.0-287                            \_SB_.PC00.RP07.PXSX.INFO
+    RP0/info.1                          0.4.7-122                            \_SB_.PC00.RP01.PXSX.INFO
+    RP0/info.2                          0.2.1-247                            \_SB_.PC00.RP10.PXSX.INFO
+    RP0/info.50.auto                    10.2.0-30                            \_SB_.PC00.RP07.PXSX.P2PF
+
+root@sonic:/home/cisco# dpupwr.py on 4
+Power on DPU4 for 20 seconds
+Power cycle DPU4 and wait 10 seconds
+Rescan PCI (30 seconds)
+[52281.119280] serial 0000:8f:00.1: Couldn't register serial port c000, irq 16, type 0, error -28
+Hello Mt Fuji
+169.254.24.1
+169.254.28.1
+169.254.32.1
+169.254.36.1
+169.254.139.1
+169.254.143.1
+169.254.147.1
+169.254.151.1
+root@sonic:/home/cisco#
+```
+#### Pass/Fail Criteria
+ * Verfiy dpu powered off in platform inventory after dpu shut down
+ * Verify dpu is showing upon platform inventory after dpu powered on
+
+### 1.10 Check removal of pcie link between npu and dpu
+
+#### Steps
+ * Use command `echo 1 > /sys/bus/pci/devices/BUS_ID/remove` to remove pcie link between npu and one dpu
+ * Use command `dpulink.sh` to assing ip address
+ * Use command `echo 1 > /sys/bus/pci/rescan` to rescan pcie links
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch: Showing example of one dpu pcie link
+
+root@sonic:/home/cisco# echo 1 > /sys/bus/pci/devices/0000:1a:00.0/remove
+root@sonic:/home/cisco# ping 169.254.28.1
+PING 169.254.28.1 (169.254.28.1) 56(84) bytes of data.
+^C
+--- 169.254.28.1 ping statistics ---
+2 packets transmitted, 0 received, 100% packet loss, time 1012ms
+
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# echo 1 > /sys/bus/pci/rescan
+root@sonic:/home/cisco# dpulink.sh
+Hello Mt Fuji
+169.254.24.1
+169.254.28.1
+169.254.32.1
+169.254.36.1
+169.254.139.1
+169.254.143.1
+169.254.147.1
+169.254.151.1
+root@sonic:/home/cisco# ping 169.254.28.1
+PING 169.254.28.1 (169.254.28.1) 56(84) bytes of data.
+64 bytes from 169.254.28.1: icmp_seq=1 ttl=64 time=0.329 ms
+^C
+--- 169.254.28.1 ping statistics ---
+1 packets transmitted, 1 received, 0% packet loss, time 0ms
+rtt min/avg/max/mdev = 0.329/0.329/0.329/0.000 ms
+root@sonic:/home/cisco# 
+
+```
+#### Pass/Fail Criteria
+ * Verfiy ping is not going through after removing pcie link.
+ * Verify ping works between dpu and npu after bringing back up the link
+
+### 1.11 Check graceful restart of NPU
+
+#### Steps
+ * Use command `reboot` to get reboot the host
+ * Wait for NPU to come up.
+ * Use command `show interface status` to chece the interfaces are up
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:/home/cisco# reboot
+.
+.
+.
+.
+root@sonic:/home/cisco# show interface status
+```
+#### Pass/Fail Criteria
+ * Verfiy all the interface are up.
```

Comment:
typo and plural

Reply:
Should we give a time expectation for this?

Reply:
corrected typo. 
yes we should give it a time for this case. That is implied.